lib/check/whitelist.js

```diff
@@ -28,5 +28,14 @@ function whitelist(ctx) {
           }
       }
     }
+    for (var whitelistedFile of jsonData.files) {
+      // Match on a substring of the reference with the whitelisted file
+      if (reference.includes(whitelistedFile)) {
+        console.warn("[Warning] Ignoring file in whitelisted link to: '%s'", whitelistedFile)
+        if (!delete references[reference]) {
+          console.error("Failed to delete the reference: '%s'", reference)
+        }
+      }
+    }
```

Comment:
I don't know anything about the context, but this looks to me to be in the wrong place.
Maybe you wanted to put it in the for loop over references?
Otherwise, the reference variable is going to be the latest value in the iteration above (if any).

Reply:
This is quite new to me too... but I think it is in the correct place.

We have one array of references -> for each of them we see if there is a match in any of the `paths` field or in any of the `files` field on the whitelist. Maybe [this commit](https://github.com/mendersoftware/mender-docs/pull/1437/commits/151af9ccc7ea07dd9adcd6d5aac89aeffb4a0c0b) helps?